src/Http/Client/Adapter/Stream.php

```diff
@@ -114,14 +115,14 @@ public function createResponses($headers, $content)
      * @param array $options Additional request options.
      * @return void
      */
-    protected function _buildContext(Request $request, $options)
+    protected function _buildContext(Request $request, array $options): void
     {
         $this->_buildContent($request, $options);
         $this->_buildHeaders($request, $options);
         $this->_buildOptions($request, $options);
 
         $url = $request->getUri();
-        $scheme = parse_url($url, PHP_URL_SCHEME);
+        $scheme = parse_url((string)$url, PHP_URL_SCHEME);
```

Comment:
This makes me sad. When a class has `__toString()` method explicit type casting shouldn't be necessary.

Reply:
How do you know it has __toString() What if the implementation of the object changes? We can ensure this method exists on the object.